src/gwt/src/org/rstudio/core/client/widget/RStudioDataGrid.java

```diff
@@ -81,5 +82,23 @@ protected void onLoad()
          
       }
       
+      // GWT's DataGrid also occasionally displays unwanted horizontal
+      // scroll bars. Since we don't ever require horizontal scrolling
+      // in the DataGrid objects we show, we just suppress horizontal
+      // overflow in any div element that does something with overflow.
+      //
+      // Again, this is something we'd normally do with CSS but because
+      // we need to perform surgery on arbitrary sub-divs which otherwise
+      // have no class we must resort to JavaScript + inline styles.
+      //
+      // https://github.com/rstudio/rstudio/issues/4529
+      children = DomUtils.querySelectorAll(parent, "div[style*='overflow:']");
```

Comment:
Do we still need to surgically remove nodes from the DOM after this change? IIRC all of the flashing scrollbars we are trying to remove by surgical element removal are horizontal, so if we're suppressing horizontal overflow everywhere that may no longer be necessary.

Reply:
Seems like it is no longer needed :-)